server/odc-service/src/main/java/com/oceanbase/odc/service/connection/ConnectionTestService.java

```diff
@@ -65,6 +67,9 @@ public ConnectionTestResult test(@NotNull @Valid TestConnectionReq req) {
         Long connectionId = req.getId();
         if (ConnectionAccountType.SYS_READ == req.getAccountType()) {
             req.setTenantName("sys");
+        } else if (StringUtils.equals("sys", req.getTenantName()) && req.getType() != ConnectType.OB_MYSQL) {
+            return ConnectionTestResult.fail(ErrorCodes.ConnectionDatabaseTypeMismatched,
```

Comment:
`ConnectionTestResult.connectTypeMismatch(req.getType())`

Reply:
q1:It doesn't have any special meaning. I put it in the if else statement before, but now I'm just moving it out
q2:why "req.getType()",i see it means target type instead of choose type
`ConnectType connectType = ConnectTypeUtil.getConnectType(
                    connectionExtensionPoint.generateJdbcUrl(jdbcUrlProperties),
                    testConnectionProperties, queryTimeoutSeconds);
            ConnectionTestResult testResult = new ConnectionTestResult(result, connectType);
            if (type != null && connectType != null && !Objects.equals(connectType, type)) {
                return ConnectionTestResult.connectTypeMismatch(connectType);
            }`